android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothStandardWeightProfile.java

```diff
@@ -823,29 +825,59 @@ protected void chooseExistingScaleUser(Vector<ScaleUser> userList) {
         chooseScaleUserUi(choices);
     }
 
-    protected String getInitials(String fullName) {
-        if (fullName == null || fullName.isEmpty() || fullName.chars().allMatch(Character::isWhitespace)) {
-            return getDefaultInitials();
-        }
-        return buildInitialsStringFrom(fullName).toUpperCase();
+    protected String getInitials(final String fullName, final int size) {
+        if (0 == size)
+            return StringUtils.EMPTY;
+
+        if (StringUtils.isAnyBlank(fullName))
+            return getDefaultInitials(size);
+
+        return buildInitialsStringFrom(fullName, size).toUpperCase();
     }
 
-    private String getDefaultInitials() {
+    private String getDefaultInitials(final int size) {
+        if (1 > size)
+            return StringUtils.EMPTY;
+
```

Comment:
In my opinion this check is redundant - this function will generate empty string if `size` is zero.
If `size` is below zero (negative), then we should allow StringBuilder to throw exception.
```suggestion
```

Reply:
It is redundant with calling method, but if other piece of code would call this one, it will behave as expected. So, if any, I would at most remove the redundancy from calling method (but I would not remove from there either as stated in related conversation)

Reply:
`size` argument comes from extension class within the same project (i.e. this method is not part of library) - if it is negative (or zero) then there is something wrong with caller code inside extension class - therefore we probably want to `fail early` to detect bug in new scale implementation (instead of continuing on with invalid arguments).